### ISSUE TITLE
Nested update fails, on one way and many to many associations

### ISSUE THREAD

**Issue by Zaggen** [balderdashy/waterline#1143]
Nested updates works as long as the nested object is associated to the 'parent' via one-to-one or many-to-one, meaning that the nested model should use the model type to define its association, or one-way associations(nested model) where the parent has a direct relationship with the nested object, meaning that the parent defines the nested model in its attributes, but the nested model don't, on any other configuration it doesn't even trigger the update on the nested model.
The specs itself doesn't state the possibility of nested updates on one way associations(nested collection) and many-to-many.
[test/unit/query/query.update.nested.js](https://github.com/balderdashy/waterline/blob/master/test/unit/query/query.update.nested.js)

Oversimplified Example using sails.js model definition (Object literal for simplification):
```js
Var Note = {
    attributes: {
      text: {type:'text'}
      images: {
        collection: 'Image'
        // No via keyword for one-way, but even with it, if the image collection
        // defines its relationship to Notes as a collection instead of a model, it
        // won't work either.
      }
   }
};

// Nested Model
Var Image = {
    attributes: {
      path:    {type: 'string'}
      width:   {type: 'int'}
      height:  {type: 'int'}
      caption: {type: 'string'}
   }
};

// Assuming the Note 1 and Image 24 already exist...
Note.update(1,  { text: 'foo', images: [{ id: 24, caption: 'At Paris' }] }).exec ....
```

If we define a many-to-many relationship between our main model(waterline collection) and the nested  model, same behavior applies. I hope you can replicate this issue, and its not just me doing something wrong. I consider this a bug, and one hard to catch, since you won't get an error, just that the nested record won't update, it'll stay the same.

I'm able to get this behavior using a custom nestedUpdate method, so it should be possible to modify the default behavior.

**Comment by particlebanana** [comment]
Here Image has no relationship to Note. In waterline the `collection` is a virtual attribute that will find child records that contain a foreign key. Changing your models to the following should fix your issues.

```javascript
Var Note = {
    attributes: {
      text: {type:'text'}
      images: {
        collection: 'Image',
        via: 'note'
      }
   }
};

// Nested Model
Var Image = {
    attributes: {
      path:    {type: 'string'}
      width:   {type: 'int'}
      height:  {type: 'int'}
      caption: {type: 'string'},
      note: { model: 'note' }
   }
};

```

**Comment by Zaggen** [comment]
@particlebanana Indeed, i know there is no relation from image to the Note model,  and i don't want to, that's why the 'One way association thing', i even stated that it works like that in the issue, but if i'm calling update on the parent it shouldn't matter, since it has all the info to update the nested models. It works for `.create`, why shouldn't work on `update`. In fact it works if i want to add a new nested record on update (when i don't provide the PK), it is expected for us the waterline users that it updates the nested associated records, not only the ones that have a one-to-many relationship, it can be done, so i don't think this issue should be marked as closed, unless that it actually works and i'm not correctly testing it on my machine.

**Comment by particlebanana** [comment]
I think the issue here is that the via-less `collection` in the `Note` model is actually creating a weird hidden join table that is being used when you do the nested adds. Then the nested updates apparently don't know how to map it correctly back to the original child record to update it. 

I'll re-open this as I guess it's a bug in that you don't get a proper explanation of why it's failing. This is a pretty loose data model though that's relying on a lot of hidden things. It might be best to create a manual mapping table to hold relationships between Note and Images that you can actually define.

**Comment by sailsbot** [comment]
Thanks for posting, @Zaggen.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
